lib/Asm89/Twig/CacheExtension/Node/CacheNode.php

```diff
@@ -32,6 +33,22 @@ public function __construct(\Twig_Node_Expression $annotation, \Twig_Node_Expres
         parent::__construct(array('key_info' => $keyInfo, 'body' => $body, 'annotation' => $annotation), array(), $lineno, $tag);
     }
 
+    private function hashIncludeNode($node, $env)
+    {
+        if ($node === null) {
+            return;
+        }
+
+        if ($node->getNodeTag() === 'include') {
+            $template = $env->getLoader()->getSource($node->getNode('expr')->getAttribute('value'));
```

Comment:
Given that this code doesn't get called when the template changes, do we even need to load and hash the full source?